src/mlpack/methods/randomized_svd/randomized_svd.cpp

```diff
@@ -47,6 +47,91 @@ RandomizedSVD::RandomizedSVD(const size_t iteratedPower,
   /* Nothing to do here */
 }
 
+void RandomizedSVD::Apply(const arma::sp_mat& data,
```

Comment:
What do you think about using a template parameter here, to reduce the code duplication? I guess something like:

```c++
template<typename DataType>
void RandomizedSVD::Apply(const arma::sp_mat& data, arma::mat& u, arma::mat& v, const size_t rank)
{
...
}

```

could work, and introduce a specialization for ``arma::sp_mat`` and ``arma::mat`` to set ``eps``. I guess, enable_if could also work to set the correct value for ``eps``.

Reply:
Thanks for the review, although using template parameter was the reason i left it as WIP :D . Couldn't get to it soon enough. The current `eps` value is working for both as of now, but its a good point to try, maybe there could be an RMSE improvement.

Reply:
@zoq I have a small doubt regarding what you mentioned. Won't using specialization or enable_if for `eps` both lead to the same function repeating with change in only one calculation, i.e similar code duplication as overloading? It might be trivial and I guess I am understanding something wrong. But any help would be great.

Reply:
I had something like this in mind:

```c++
template<typename DataType>
void RandomizedSVD::Apply(const arma::mat& data, arma::mat& u, arma::mat& v, const size_t rank, const double eps)
{
...
}

void RandomizedSVD::Apply(const arma::sp_mat& data, arma::mat& u, arma::mat& v, const size_t rank)
{
  Apply(data, u, v, rank, 0);
}

void RandomizedSVD::Apply(const arma::mat& data, arma::mat& u, arma::mat& v, const size_t rank)
{
  Apply(data, u, v, rank, eps);
}
```

Let me know what you think.

Reply:
Sorry for the late response. That did help understand what you meant. Thank you. :)

Reply:
No worries, as you said this is a WIP PR, and I don't think 10 hours counts as late 😃